accepted/2024/NuGet-Audit-for-PackageDownload.md

```diff
@@ -0,0 +1,106 @@
+- [GitHub Issue](https://github.com/NuGet/Home/issues/13658)
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+In our effort to strengthen the security of NuGet, we would like to expand the support of NuGet Audit to `<PackageDownload>` packages. This will be done by introducing vulnerability errors for packages downloaded using `PackageDownload`. However, the .Net sdk also makes use of `PackageDownload` to implicitly download packages. And we would like to make sure these packages are not also receiving these errors, in order to prevent customers being confused. This will be done by introducing a new attribute to `PackageDownload` that makes it clear to NuGet Audit it should not write an error for these packages.
+
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+
+We are expanding NuGet Audit to `<PackageDownload>` packages to improve security notifications for user-specified dependencies, addressing gaps in vulnerability alerts that currently miss these packages. This change prevents confusion by excluding SDK-managed PackageDownloads from error reports, ensuring users only see relevant security issues. The expected outcome is a more precise and useful vulnerability reporting experience, enabling users to address actual risks without unnecessary SDK-related alerts.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+The NuGet Audit now supports `<PackageDownload>` packages, allowing it to detect and alert users about vulnerabilities in any user-specified dependencies downloaded via `<PackageDownload>`. This means you’ll now receive warnings if a package added via `<PackageDownload>` has known security issues, so you can quickly address them.
+
+However, the .NET SDK itself also uses `<PackageDownload>` to pull in SDK-specific dependencies, and to avoid confusing users, these SDK-managed packages are excluded from NuGet Audit warnings. We achieve this using a new attribute on `<PackageDownload>` entries in the SDK, such as `IsImplicitlyDefined="true"`. When NuGet Audit sees this attribute, it knows to skip auditing this package.
+
+#### Real-Life Example
+
+Imagine you have a project that downloads some libraries with `<PackageDownload>`:
+
+```xml
+<ItemGroup>
+    <PackageDownload Include="NuGet.Protocol" Version="[5.11.2]" />
+</ItemGroup>
+```
+
+If "NuGet.Protocol" is found to have a vulnerability, NuGet Audit will issue a warning to alert you so you can update to a safer version.
+
+##### Warning NU1906
+
+> warning NU1906: Package 'NuGet.Protocol' 5.11.2 has a known moderate severity vulnerability, https://github.com/advisories/GHSA-g3q9-xf95-8hp5
```

Comment:
Did you consider using the same warning codes as PackageDownload?
We don't have strict rules around the warning codes, but we do it based off the UX we'd want people to have. 
Would we have situations in which people want to do something about PackageDownload but not about PackageReference or vice versa? Would the suppressions functionality cover that gap?